File: src/types/photo.ts
// Defines the structure of a Photo object with id, title, and url properties.
type Photo = {
  id: number;
  filename: string;
  stored_filename: string;
  mime_type: string;
  file_size: number;
  title: string | null;
  description: string | null;
  is_active: boolean;
  sort_order: number;
  created_at: string;
  updated_at: string;
  image_url: string;
};

export default Photo;

File: src/schemas/photo.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PhotoResponse(BaseModel):
    id: int
    filename: str
    stored_filename: str
    mime_type: str
    file_size: int
    title: str | None
    description: str | None
    is_active: bool
    sort_order: int
    created_at: datetime
    updated_at: datetime
    image_url: str

    # from Collection
    collection_ids: list[int] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)


class PhotoUpdateRequest(BaseModel):
    title: str | None = None
    description: str | None = None
    is_active: bool | None = None
